### ISSUE TITLE
Question on comparing overlap of two segments: How to handle floating point errors.

### ISSUE THREAD

**Issue by gigerdo** [alexbol99/flatten-js#60]
The setup:
- I have two rectangles(as polygons) that have the same rotation, but have different sizes.
- The rectangles are touching and I want to move them so the length of the touching segment is maximized.
- I do this by calculating the intersection of the touching sides (as segments).

The problem:
Because of floating point calculations, the rectangles are sometimes slightly apart and it is not possible to directly calculate the overlap.

Do you have any idea how to make this more robust? 
One of my ideas was to use the flatten/offset library to make the rectangles or the side-segments larger and then calculate the intersection on those. This should make the calculation more stable.

**Comment by gigerdo** [comment]
I tried my idea and I have another problem there. The intersection crashes seemingly if the polygons match nearly, but not close enough. If I slightly change the points it suddenly works. Otherwise I get the error below.

```
import Flatten from "@flatten-js/core";
import Polygon = Flatten.Polygon;
import point = Flatten.point;
import BooleanOperations = Flatten.BooleanOperations;

const poly1 = new Polygon([
    point(709.6637, 335.5266),
    point(809.2982, 326.9851),
    point(808.4441, 317.0216),
    point(708.8095, 325.5631),
]);

const poly2 = new Polygon([
    point(838.3345, 314.4592),
    point(738.6999, 323.0007),
    point(739.5541, 332.9641),
    point(839.1886, 324.4227),
]);

const intersection = BooleanOperations.intersect(poly1, poly2);
```

Error:
```
Error: Unresolved boundary conflict in boolean operation
    at Function.get UNRESOLVED_BOUNDARY_CONFLICT [as UNRESOLVED_BOUNDARY_CONFLICT] (project\node_modules\@flatten-js\core\dist\main.cjs.js:154:16)
    at fixBoundaryConflicts (project\node_modules\@flatten-js\core\dist\main.cjs.js:1031:34)
    at filterNotRelevantEdges (project\node_modules\@flatten-js\core\dist\main.cjs.js:498:12)
    at booleanOpBinary (project\node_modules\@flatten-js\core\dist\main.cjs.js:555:5)
    at Object.intersect (project\node_modules\@flatten-js\core\dist\main.cjs.js:408:32)
    at Object.<anonymous> (project\scratch.ts:20:40)
    at Module._compile (internal/modules/cjs/loader.js:1156:30)
    at Module.m._compile (project\node_modules\ts-node\src\index.ts:1043:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:1176:10)
    at Object.require.extensions.<computed> [as .ts] (project\node_modules\ts-node\src\index.ts:1046:12)

Process finished with exit code 1
```

**Comment by alexbol99** [comment]
Indeed, floating point calculation is not robust.
I you know that your shapes are rectangles, you may try to rotate them so that they become axes aligned, calculate intersections and then rotate them back.

**Comment by alexbol99** [comment]
Hi, @gigerdo ,
The way I suggested does not work.
Please send me an example of what your want to get, I will try to find out some workaround
B.R.,
Alex

**Comment by gigerdo** [comment]
Hi @alexbol99 
Thanks a lot for having a look at it.

Here is an observable with some examples:
https://observablehq.com/@giger/calculating-the-length-of-the-side-touching-between-two-rec

Unfortunately I forgot to save any examples where the calculation fails because of floating point errors. The problem was that sometimes the segments are not on top of each other anymore because of a slight error. 

Maybe you have an idea on how to solve this in a more robust manner?

**Comment by gigerdo** [comment]
I think I found another example where a problem with floating point errors occur, maybe you can help me with this one:

https://observablehq.com/@giger/flattenjs-infinite-loop-unify

BooleanOperations.unify on these polygons runs into an infinite-loop error. Do you have an idea how this might be solved?

**Comment by gigerdo** [comment]
The problem from my previous post was also fixed with 1.2.22. So I am going to close this ticket.
